lib/dgram.js

```diff
@@ -272,6 +267,41 @@ Socket.prototype.send = function(buffer,
   if (offset + length > buffer.length)
     throw new RangeError('Offset + length beyond buffer length');
 
+  return buffer.slice(offset, offset + length);
+}
+
+Socket.prototype.send = function(buffer,
+                                 offset,
+                                 length,
+                                 port,
+                                 address,
+                                 callback) {
+  var self = this;
+
+  if (arguments.length === 5 || arguments.length === 6) {
+    buffer = sliceBuffer(buffer, offset, length);
```

Comment:
I feel like it could be potentially slower and will do excessive allocation. Perhaps we should just pass offset to the C++ layer?